std/algorithm/sorting.d

```diff
@@ -2020,8 +2020,8 @@ unittest
         bool proxySwapCalled;
         struct S
         {
-            int i;
-            alias i this;
+            int m;
+            alias m this;
```

Comment:
What is the problem that this is trying to fix? This variable doesn't conflict with anything. It's in a different scope from the other `i`, so I really don't see the problem. And IMHO, if a change needs to be made, the preceding variable should have braces put around the tests that it's in rather than renaming this one, since the other one exists way beyond where it's needed.

But aside from this specific example, I don't understand what the problem is that we're trying to fix with this PR. It seems like we're trying to "fix" warnings about completely legitimate code.